typespec-tests/src/main/java/com/cadl/patch/models/Fish.java

```diff
@@ -24,43 +24,48 @@ public class Fish implements JsonSerializable<Fish> {
      * The kind property.
      */
     @Generated
-    private String kind;
+    String kind;
 
     /*
      * The id property.
      */
     @Generated
-    private String id;
+    String id;
 
     /*
      * The name property.
      */
     @Generated
-    private String name;
+    String name;
```

Comment:
Using `package-private` for the properties and remove the setters in the `fromJson` can solve the issue. But this could leave chance for user to inherit this model, would like to hear @srnagar's thoughts as well.

Reply:
Making it package private likely going to cause problem, when class be in different packages...

Reply:
Since it seems we've aimed to make json-merge-patch an implementation detail, even if we use the access helper pattern that'd still be internal only. So, we may be at a point where we either continue with internal only and don't allow customer extension or we need to start adding APIs to the public surface area to support extension.